Backend/src/Trackable.Repositories/Repositories/TrackingPointRepository.cs

```diff
@@ -43,26 +43,32 @@ public TrackingPointRepository(TrackableDbContext db, IMapper mapper)
                 return new List<TrackingPoint>();
             }
 
-            string deviceId = models.First().TrackingDeviceId;
-
-            var device = await this.Db.TrackingDevices
-                .Where(d => !d.Deleted)
-                .Include(d => d.Asset)
-                .SingleAsync(d => d.Id == deviceId);
-
-            if (device.Asset == null)
+            var devicePointsLookup = models.ToLookup(m => m.TrackingDeviceId);
+            var savedModels = new List<TrackingPoint>();
+            foreach (var dpl in devicePointsLookup)
             {
-                throw new InvalidOperationException("Can't add a tracking point while device not linked to an asset");
+                var device = await this.Db.TrackingDevices
+                   .Where(d => !d.Deleted)
+                   .Include(d => d.Asset)
+                   .SingleAsync(d => d.Id == dpl.Key);
+
+                if (device.Asset == null)
+                {
+                    throw new InvalidOperationException("Can't add a tracking point while device not linked to an asset");
```

Comment:
Consider the case where one of the devices, does not have a linked asset, the request would partially fail. Some of the points will be added and some won't. We should fail the request and not add any points in this case.